### PR TITLE
Add {compressed}tex{Sub}Image*D entry points with server side data.

### PR DESCRIPTION
**Pull request by zhenyao**
@kenrussell @jdashg

### CODE REVIEW

**Comment by kenrussell** [comment]
Minor nit: "service" should be "server" here, in the terminology of the OpenGL client/server model. Would you mind making that change?

File: specs/latest/2.0/index.html [hunk 1]
```diff
@@ -1340,14 +1342,25 @@ <h2 class="no-toc">Table of contents</h2>
...
+        <p>If an attempt is made to call the function with no WebGLTexture bound, generates an <code>INVALID_OPERATION</code> error.</p>
+        <p>If no WebGLBuffer is bound to the <code>PIXEL_UNPACK_BUFFER</code> target, generates an <code>INVALID_OPERATION</code> error.</p>
+      </dd>
+
       <dt class="idl-code"><a name="TEXSUBIMAGE2D">void texSubImage2D</a>(GLenum target, GLint level, GLint xoffset, GLint yoffset, GLsizei width, GLsizei height, GLenum format, GLenum type, ArrayBufferView? pixels)
         <span class="gl-spec">(<a href="http://www.khronos.org/registry/gles/specs/3.0/es_full_spec_3.0.3.pdf#nameddest=section-3.8.5">OpenGL ES 3.0.3 &sect;3.8.5</a>, <a class="nonnormative" href="https://www.khronos.org/opengles/sdk/docs/man3/html/glTexSubImage2D.xhtml">man page</a>)</span>
       </dt>
       <dd>
         <p>Only differences from <a href="https://www.khronos.org/registry/webgl/specs/latest/1.0/#TEXSUBIMAGE2D">texSubImage2D</a> in WebGL 1 are described here. </p>
+        <p>If a WebGLBuffer is bound to the <code>PIXEL_UNPACK_BUFFER</code> target, generates an <code>INVALID_OPERATION</code>. error</p>
```
Comment (kenrussell) [inline specs/latest/2.0/index.html]: Period's in the wrong place on this line.

File: specs/latest/2.0/index.html [hunk 2]
```diff
@@ -1361,14 +1374,26 @@ <h2 class="no-toc">Table of contents</h2>
...
         <p>The combination of <em>format</em>, <em>type</em>, and WebGLTexture's internal format must be listed in Table 1 or 2 from <span class="gl-spec"><a class="nonnormative" href="https://www.khronos.org/opengles/sdk/docs/man3/html/glTexImage2D.xhtml">man page</a></span>.</p>
       </dd>
 
+      <dt><p class="idl-code"><a name="TEXSUBIMAGE2D_SERVICE_SIDE">void texSubImage2D</a>(GLenum target, GLint level, GLint xoffset, GLint yoffset, GLenum format, GLenum type, GLintptr offset)
+        <span class="gl-spec">(<a href="http://www.khronos.org/registry/gles/specs/3.0/es_full_spec_3.0.3.pdf#nameddest=section-3.8.5">OpenGL ES 3.0.3 &sect;3.8.5</a>, <a class="nonnormative" href="https://www.khronos.org/opengles/sdk/docs/man3/html/glTexSubImage2D.xhtml">man page</a>)</span>
+      </dt>
+      <dd>
+        <p>Updates a sub-rectangle of the currently bound WebGLTexture with data from the WebGLBuffer bound to <code>PIXEL_UNPACK_BUFFER</code> target.</p>
+        <p><em>offset</em> is the byte offset into the WebGLBuffer's data store; generates an <code>INVALID_VALUE</code> if it's less than 0.</p>
```
Comment (kenrussell) [inline specs/latest/2.0/index.html]: INVALID_VALUE -> INVALID_VALUE error

Comment (kenrussell) [inline specs/latest/2.0/index.html]: (Here and throughout)

**Comment by kenrussell** [comment]
Very nice overall. Could you make the one other minor change about INVALID_VALUE errors? Please feel free to merge yourself afterward. LGTM

**Comment by kenrussell** [comment]
Also, possible to make the service -> server change in the pull request's description too?

**Comment by zhenyao** [comment]
Thanks for the quick review.  All comments addressed.

**Pull request merged**